src/main.cpp

```diff
@@ -906,7 +911,19 @@ static int main0(int argc, char **argv) {
         }
         if (c_source_files.length == 0 && objects.length == 0) {
             // For example `zig cc` and no args should print the "no input files" message.
-            return ZigClang_main(argc, argv);
+            if(zig_only_argc) {
+                ZigList<const char *> clang_argv = {0};
+                for (int i = 0; i < argc; i += 1) {
+                    if(strcmp(argv[i], "--cache-dir") == 0) {
+                        i += 1;
+                    } else {
+                        clang_argv.append(argv[i]);
+                    }
+                }
+                return ZigClang_main(clang_argv.length, (char **)clang_argv.items);
```

Comment:
I'm still unhappy about this section. I'm not sure if ZigList<const char *> is the type I should use for this and I don't like the fact that I have to use this (char **) cast to avoid an invalid conversion error.